### PR TITLE
Fix possibly unefined $form variables in smarty templates

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes a bunch of e-notices coming from smarty templates using mass regex find & replace.

Technical Details
----------------------------------------
Referencing a property of the $form in smarty is almost always going to be undefined if it hasn't been set on the form, so this fixes a bad pattern of not using `empty()` to guard against e-notices.

### CODE REVIEW

**Comment by colemanw** [comment]
In general I agree that mass-updating templates with `!empty()` could mask some underlying problems. However in the case of accessing `$form` elements, they are created via `CRM_Core_Form::addElement` or `addField` methods rather than through `$this->assign()` so 9 times out of 10 if the template says `{if $form.foo}` it's a lazy/sloppy substitute for `{if !empty($form.foo)}`.

**Comment by eileenmcnaughton** [comment]
Yeah - I'm OK with this - I'm getting a bit fed up with these notices & I agree that these are a bit less likely to be completely obsolete than assigns

File: ext/eventcart/templates/CRM/Event/Cart/Form/Checkout/Payment.tpl [hunk 1]
```diff
@@ -94,7 +94,7 @@
   </tfoot>
```
Comment (eileenmcnaughton) [inline ext/eventcart/templates/CRM/Event/Cart/Form/Checkout/Payment.tpl:94]: Checked this file = ok

File: ext/legacycustomsearches/templates/CRM/Contact/Form/Search/Custom/MultipleValuesCriteria.tpl [hunk 2]
```diff
@@ -29,7 +29,7 @@
           <div class="clear"></div>
```
Comment (eileenmcnaughton) [inline ext/legacycustomsearches/templates/CRM/Contact/Form/Search/Custom/MultipleValuesCriteria.tpl:29]: Checked this file = ok

File: templates/CRM/Activity/Form/Search/Common.tpl [hunk 3]
```diff
@@ -14,7 +14,7 @@
   </td>
 </tr>
```
Comment (eileenmcnaughton) [inline templates/CRM/Activity/Form/Search/Common.tpl:15]: Checked this file = ok

File: templates/CRM/Admin/Form/PaymentProcessor.tpl [hunk 4]
```diff
@@ -58,37 +58,37 @@
         <tr class="crm-paymentProcessor-form-block-user_name">
             <td class="label">{$form.user_name.label}</td><td>{$form.user_name.html} {help id=$ppTypeName|cat:'-live-user-name' title=$form.user_name.label}</td>
         </tr>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/PaymentProcessor.tpl:60]: Checked this file = ok

File: templates/CRM/Admin/Form/Preferences/Display.tpl [hunk 5]
```diff
@@ -212,7 +212,7 @@
   </table>
   <div class="crm-submit-buttons">{include file="CRM/common/formButtons.tpl" location="bottom"}</div>
 </div>
-{if $form.contact_edit_options.html}
+{if !empty($form.contact_edit_options.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/Preferences/Display.tpl:215]: Checked this file = ok

File: templates/CRM/Admin/Form/PreferencesDate.tpl [hunk 6]
```diff
@@ -20,7 +20,7 @@
             <tr class="crm-preferences-date-form-block-date_format">
                 <td class="label">{$form.date_format.label}</td><td>{$form.date_format.html}</td>
             </tr>
-            {if $form.time_format.label}
+            {if !empty($form.time_format.label)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/PreferencesDate.tpl:23]: Checked this file = ok

File: templates/CRM/Admin/Form/Setting/Localization.tpl [hunk 7]
```diff
@@ -19,7 +19,7 @@
                 <td class="label">{$form.lcMessages.label}</td>
                 <td>{$form.lcMessages.html}</td>
             </tr>
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/Setting/Localization.tpl:21]: Checked this file = ok

File: templates/CRM/Admin/Form/Setting/UF.tpl [hunk 8]
```diff
@@ -17,7 +17,7 @@
             <td class="label">{$form.userFrameworkUsersTableName.label}</td>
             <td>{$form.userFrameworkUsersTableName.html}</td>
         </tr>
-        {if $form.wpBasePage}
+        {if !empty($form.wpBasePage)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Admin/Form/Setting/UF.tpl:20]: Checked this file = ok

File: templates/CRM/Case/Form/Activity.tpl [hunk 9]
```diff
@@ -233,7 +233,7 @@
         </td>
       </tr>
     {/if}
-    {if $form.tag.html}
+    {if !empty($form.tag.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Case/Form/Activity.tpl:236]: Checked this file = ok

File: templates/CRM/Case/Form/ActivityTab.tpl [hunk 10]
```diff
@@ -45,7 +45,7 @@
             {$form.activity_type_filter_id.html}
           </td>
         </tr>
-        {if $form.activity_deleted}
+        {if !empty($form.activity_deleted)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Case/Form/ActivityTab.tpl:48]: Checked this file = ok

File: templates/CRM/Case/Form/Case.tpl [hunk 11]
```diff
@@ -53,7 +53,7 @@
     </tr>
 {/if}
 
-{if $form.activity_details.html}
+{if !empty($form.activity_details.html)}
     <tr class="crm-case-form-block-activity_details">
```
Comment (eileenmcnaughton) [inline templates/CRM/Case/Form/Case.tpl:57]: Checked this file = ok

File: templates/CRM/Case/Form/Search/Common.tpl [hunk 12]
```diff
@@ -41,7 +41,7 @@
         <br />
         {$form.case_owner.html}
       {/if}
-      {if $form.case_deleted}
+      {if !empty($form.case_deleted)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Case/Form/Search/Common.tpl:44]: Checked this file = ok

File: templates/CRM/Contact/Form/Edit/Individual.tpl [hunk 13]
```diff
@@ -10,7 +10,7 @@
 {* tpl for building Individual related fields *}
 <table class="form-layout-compressed">
   <tr>
-    {if $form.prefix_id}
+    {if !empty($form.prefix_id)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Edit/Individual.tpl:13]: Checked this file = ok

**Comment by eileenmcnaughton** [comment]
I can't keep myself focussed on checking this many changes - I've commented where I have checked so far

File: templates/CRM/common/CMSUser.tpl [hunk 14]
```diff
@@ -11,7 +11,7 @@
    <fieldset class="crm-group crm_user-group">
```
Comment (eileenmcnaughton) [inline templates/CRM/common/CMSUser.tpl:11]: Checked this file = ok

File: templates/CRM/Price/Form/Option.tpl [hunk 15]
```diff
@@ -63,14 +63,14 @@
         </td>
       </tr>
       {* fix for CRM-10241 *}
-      {if $form.count.html}
+      {if !empty($form.count.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Price/Form/Option.tpl:66]: Checked this file = ok

File: templates/CRM/Pledge/Form/Pledge.tpl [hunk 16]
```diff
@@ -81,7 +81,7 @@
           </tr>
 
         {if $email and $outBound_option != 2}
-            {if $form.is_acknowledge }
+            {if !empty($form.is_acknowledge)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Pledge/Form/Pledge.tpl:84]: Checked this file = ok

File: templates/CRM/Member/Form/Search/Common.tpl [hunk 17]
```diff
@@ -37,7 +37,7 @@
 </tr>
 <tr>
   <td>
-    {if $form.member_auto_renew}
+    {if !empty($form.member_auto_renew)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Form/Search/Common.tpl:40]: Checked this file = ok

File: templates/CRM/Member/Form/MembershipBlock.tpl [hunk 18]
```diff
@@ -12,7 +12,7 @@
 <div class="help">
     {ts}Use this form to enable and configure a Membership Signup and Renewal section for this Online Contribution Page. If you're not using this page for membership signup, leave the <strong>Enabled</strong> box un-checked..{/ts} {docURL page="user/membership/setup"}
 </div>
-  {if $form.membership_type.html}
+  {if !empty($form.membership_type.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Member/Form/MembershipBlock.tpl:15]: Checked this file = ok

File: templates/CRM/Mailing/Form/Search.tpl [hunk 19]
```diff
@@ -26,7 +26,7 @@
           {$form.is_archived.html}
         </div>
       </td>
-      {if $form.mailing_status}
+      {if !empty($form.mailing_status)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Mailing/Form/Search.tpl:29]: Checked this file = ok

File: templates/CRM/Group/Form/Edit.tpl [hunk 20]
```diff
@@ -45,7 +45,7 @@
       <td>{$form.frontend_description.html}</td>
     </tr>
 
-    {if $form.group_type}
+    {if !empty($form.group_type)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Group/Form/Edit.tpl:48]: Checked this file = ok

File: templates/CRM/Form/body.tpl [hunk 21]
```diff
@@ -7,11 +7,11 @@
  | and copyright information, see https://civicrm.org/licensing       |
  +--------------------------------------------------------------------+
 *}
-{if $form.javascript}
+{if !empty($form.javascript)}
   {$form.javascript}
 {/if}
 
-{if $form.hidden}
+{if !empty($form.hidden)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Form/body.tpl:14]: Checked this file = ok

File: templates/CRM/Form/attachment.tpl [hunk 22]
```diff
@@ -41,7 +41,7 @@
      {/if}
     <div id="attachments">
       <table class="form-layout-compressed">
-      {if $form.attachFile_1}
+      {if !empty($form.attachFile_1)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Form/attachment.tpl:44]: Checked this file = ok

File: templates/CRM/Contact/Form/Inline/ContactName.tpl [hunk 23]
```diff
@@ -14,37 +14,37 @@
     {include file="CRM/common/formButtons.tpl"}
   </div>
   {if $contactType eq 'Individual'}
-    {if $form.prefix_id}
+    {if !empty($form.prefix_id)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Inline/ContactName.tpl:17]: Checked this file = ok

File: templates/CRM/Contact/Form/Search/ContactSearchFields.tpl [hunk 24]
```diff
@@ -7,7 +7,7 @@
   <td>{$form.buttons.html}</td>
 </tr>
 <tr>
-  {if $form.contact_tags}
+  {if !empty($form.contact_tags)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Search/ContactSearchFields.tpl:10]: Checked this file = ok

File: templates/CRM/Contact/Form/Search/Criteria/DisplaySettings.tpl [hunk 25]
```diff
@@ -2,11 +2,11 @@
   <table>
     <tr>
       <td>
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Search/Criteria/DisplaySettings.tpl:4]: checked this file = ok

File: templates/CRM/Contact/Form/Search/Criteria/Location.tpl [hunk 26]
```diff
@@ -72,13 +72,13 @@
           <div>{$form.location_type.label} {help id="location_type" title=$form.location_type.label}</div>
           {$form.location_type.html}
         </div>
-        {if $form.address_name.html}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Search/Criteria/Location.tpl:75]: checked this file = ok

File: templates/CRM/Contact/Form/Search/Criteria/SearchSettings.tpl [hunk 27]
```diff
@@ -3,7 +3,7 @@
     <tr>
       <td>{$form.operator.label} {help id="id-search-operator"}<br />{$form.operator.html}</td>
       <td>
-        {if $form.deleted_contacts}{$form.deleted_contacts.html} {$form.deleted_contacts.label}{/if}
+        {if !empty($form.deleted_contacts)}{$form.deleted_contacts.html} {$form.deleted_contacts.label}{/if}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Search/Criteria/SearchSettings.tpl:6]: checked this file = ok

File: templates/CRM/Contact/Form/Task/AddToGroup.tpl [hunk 28]
```diff
@@ -33,7 +33,7 @@
                 <tr class="crm-contact-task-addtogroup-form-block-description">
                    <td class="label">{$form.description.label}</td>
                    <td>{$form.description.html}</td></tr>
-                {if $form.group_type}
+                {if !empty($form.group_type)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Task/AddToGroup.tpl:36]: checked this file = ok

File: templates/CRM/Contact/Form/Task/PDFLetterCommon.tpl [hunk 29]
```diff
@@ -8,7 +8,7 @@
  +--------------------------------------------------------------------+
 *}
 {*common template for compose PDF letters*}
-{if $form.template.html}
+{if !empty($form.template.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Task/PDFLetterCommon.tpl:11]: checked this file = ok

File: templates/CRM/Contact/Form/Task/SaveSearch.tpl [hunk 30]
```diff
@@ -34,7 +34,7 @@
       <td class="label">{$form.description.label}</td>
       <td>{$form.description.html}</td>
     </tr>
-    {if $form.group_type}
+    {if !empty($form.group_type)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Form/Task/SaveSearch.tpl:37]: checked this file = ok

File: templates/CRM/Contact/Import/Form/Preview.tpl [hunk 31]
```diff
@@ -102,7 +102,7 @@
 
       {* Existing Group *}
 
-<div id="existing-groups" class="crm-accordion-wrapper crm-existing_group-accordion {if $form.groups} {else}collapsed{/if}">
+<div id="existing-groups" class="crm-accordion-wrapper crm-existing_group-accordion {if !empty($form.groups)} {else}collapsed{/if}">
```
Comment (eileenmcnaughton) [inline templates/CRM/Contact/Import/Form/Preview.tpl:105]: checked this file = ok

File: templates/CRM/Contribute/Form/Contribution.tpl [hunk 32]
```diff
@@ -228,7 +228,7 @@
           </td>
         </tr>
         {/if}
-        {if $form.payment_processor_id}
+        {if !empty($form.payment_processor_id)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Contribution.tpl:231]: checked this file = ok

File: templates/CRM/Contribute/Form/Contribution/Main.tpl [hunk 33]
```diff
@@ -135,7 +135,7 @@
       {/if}
       {/crmRegion}
 
-      {if $form.is_recur}
+      {if !empty($form.is_recur)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Contribution/Main.tpl:138]: checked this file = ok

File: templates/CRM/Contribute/Form/Contribution/Main.tpl [hunk 34]
```diff
@@ -199,7 +199,7 @@
             <div class="crm-public-form-item crm-section honor_block_text-section">
               {$honor_block_text}
             </div>
-          {if $form.soft_credit_type_id.html}
+          {if !empty($form.soft_credit_type_id.html)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Contribution/Main.tpl:202]: checked this file = ok

File: templates/CRM/Contribute/Form/Contribution/Main.tpl [hunk 35]
```diff
@@ -256,7 +256,7 @@
       {* end of ccid loop *}
     {/if}
 
-    {if $form.payment_processor_id.label}
+    {if !empty($form.payment_processor_id.label)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Contribution/Main.tpl:259]: checked this file = ok

File: templates/CRM/Financial/Form/BatchTransaction.tpl [hunk 36]
```diff
@@ -25,15 +25,15 @@
             </td>
           </tr>
           <tr>
-          {if $form.contact_tags}
+          {if !empty($form.contact_tags)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Financial/Form/BatchTransaction.tpl:28]: checked this file = ok

File: templates/CRM/Export/Form/Select.tpl [hunk 37]
```diff
@@ -29,7 +29,7 @@
   </div>
 
   <div id="map" class="crm-section crm-export-mapping-section">
-      {if $form.mapping }
+      {if !empty($form.mapping)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Export/Form/Select.tpl:32]: checked this file = ok

File: templates/CRM/Event/Form/Task/AddToGroup.tpl [hunk 38]
```diff
@@ -33,7 +33,7 @@
                 <tr class="crm-contact-task-addtogroup-form-block-description">
                    <td class="label">{$form.description.label}</td>
                    <td>{$form.description.html}</td></tr>
-                {if $form.group_type}
+                {if !empty($form.group_type)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/Task/AddToGroup.tpl:36]: checked this file = ok

File: templates/CRM/Event/Form/SelfSvcTransfer.tpl [hunk 39]
```diff
@@ -28,7 +28,7 @@
       <td class="crm-participant-participant_role">{$details.role}</td>
     </tr>
   </table>
-  {if $form.contact_id}
+  {if !empty($form.contact_id)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/SelfSvcTransfer.tpl:31]: checked this file = ok

File: templates/CRM/Event/Form/Registration/Register.tpl [hunk 40]
```diff
@@ -122,7 +122,7 @@
       </fieldset>
     {/if}
 
-    {if $form.payment_processor_id.label}
+    {if !empty($form.payment_processor_id.label)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/Registration/Register.tpl:125]: checked this file = ok

File: templates/CRM/Event/Form/ManageEvent/EventInfo.tpl [hunk 41]
```diff
@@ -15,13 +15,13 @@
     {include file="CRM/common/formButtons.tpl" location="top"}
   </div>
   <table class="form-layout-compressed">
-    {if $form.template_id}
```
Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/ManageEvent/EventInfo.tpl:18]: checked this file = ok

File: templates/CRM/Event/Form/EventFees.tpl [hunk 42]
```diff
@@ -227,7 +227,7 @@
 
      function fillTotalAmount( totalAmount ) {
           if ( !totalAmount ) {
-        var amountVal = {/literal}{if $form.amount.value}{$form.amount.value}{else}0{/if}{literal};
```
Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/EventFees.tpl:230]: checked this file = ok

File: templates/CRM/Custom/Form/Optionfields.tpl [hunk 43]
```diff
@@ -15,7 +15,7 @@
 </td>
 </tr>
 
-<tr id="option_group" {if !$form.option_group_id}class="hiddenElement"{/if}>
+<tr id="option_group" {if empty($form.option_group_id)}class="hiddenElement"{/if}>
```
Comment (eileenmcnaughton) [inline templates/CRM/Custom/Form/Optionfields.tpl:18]: checked this file = ok

File: templates/CRM/Contribute/Form/ContributionPage/Amount.tpl [hunk 44]
```diff
@@ -420,7 +420,7 @@
 
 </script>
 {/literal}
-{if $form.is_recur}
+{if !empty($form.is_recur)}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/ContributionPage/Amount.tpl:423]: checked this file = ok

File: templates/CRM/Contribute/Form/Contribution/OnBehalfOf.tpl [hunk 45]
```diff
@@ -13,7 +13,7 @@
  * FIXME: This is way more complex than it needs to be
  * FIXME: Why are we not just using the dynamic form tpl to display this profile?
  *}
-{if $form.is_for_organization}
```
Comment (eileenmcnaughton) [inline templates/CRM/Contribute/Form/Contribution/OnBehalfOf.tpl:16]: checked this file = ok

**Comment by eileenmcnaughton** [comment]
OK - I think I've eyeballed them all. I definitely think fixing enotices is a chance to fix cruft - but OTOH there are a lot to work through so gonna merge this

**Pull request merged**